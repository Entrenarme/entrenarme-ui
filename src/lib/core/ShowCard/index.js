// @flow
import * as React from 'react';
import styled from 'styled-components';
import ExpansionPanel from '@material-ui/core/ExpansionPanel';
import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';
import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faAngleDown } from '@fortawesome/free-solid-svg-icons';
import JssProvider from 'react-jss/lib/JssProvider';
import { create } from 'jss';
import {
  createGenerateClassName,
  jssPreset,
  createMuiTheme,
  MuiThemeProvider,
} from '@material-ui/core/styles';

import H4 from '../Text/Headers/H4/index';

const theme = createMuiTheme();

const generateClassName = createGenerateClassName({
  productionPrefix: 'showcard-eui',
});
const jss = create(jssPreset());

const ExtendedGlobalContainer = styled(ExpansionPanel)`
  > div {
    padding: 0px;
  }

  margin: 0px !important;
  box-shadow: none !important;
  border-radius: 2px !important;

  &:before {
    background-color: transparent !important;
  }
`;

const ExtendedTitle = styled(ExpansionPanelSummary)`
  div {
    padding: 25px 30px;
  }

  div:last-child {
    padding: 0px;
  }
`;

const ExtendedBody = styled(ExpansionPanelDetails)`
  padding: 0px !important;
`;

const BodyContainer = styled.div``;

type Props = {
  children: React.Node,
  title: string,
};

type State = {
  expanded: ?string,
};

class ShowCard extends React.Component<Props, State> {
  state = {
    expanded: null,
  };

  handleChange = (panel: string) => (event: Object, expanded: boolean) => {
    this.setState({
      expanded: expanded ? panel : false,
    });
  };

  render() {
    const { children, title, ...rest } = this.props;
    const { expanded } = this.state;

    return (
      <JssProvider jss={jss} generateClassName={generateClassName}>
        <MuiThemeProvider theme={theme}>
          <ExtendedGlobalContainer
            {...rest}
            expanded={expanded === 'panel1'}
            onChange={this.handleChange('panel1')}
          >
            <ExtendedTitle expandIcon={<FontAwesomeIcon icon={faAngleDown} />}>
              <H4 size="small" style={{ fontWeight: 600, margin: '0px' }}>
                {title}
              </H4>
            </ExtendedTitle>
            <ExtendedBody>
              <BodyContainer>{children}</BodyContainer>
            </ExtendedBody>
          </ExtendedGlobalContainer>
        </MuiThemeProvider>
      </JssProvider>
    );
  }
}

ShowCard.defaultProps = {};

export default ShowCard;
