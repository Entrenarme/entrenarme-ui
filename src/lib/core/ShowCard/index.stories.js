// @flow
import * as React from 'react';
import { storiesOf } from '@storybook/react';

import ShowCard from './index.js';
import P from '../Text/Paragraph/index';

const ElementDecoration = storyFn => (
  <div style={{ backgroundColor: '#F1F1F1', height: '100vh', padding: '20px' }}>
    {storyFn()}
  </div>
);

storiesOf('ShowCard', module)
  .addDecorator(ElementDecoration)
  .add('Show card with basic content', () => (
    <ShowCard>
      <div panel="panel1">
        <div>jajaja</div>
        <div>
          <P>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec
            fermentum magna eu velit semper hendrerit. Aenean aliquet pretium
            justo, ut rhoncus magna euismod et. Donec condimentum congue sapien,
            nec auctor eros sagittis eget.
          </P>
          <P>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec
            fermentum magna eu velit semper hendrerit. Aenean aliquet pretium
            justo, ut rhoncus magna euismod et. Donec condimentum congue sapien,
            nec auctor eros sagittis eget.
          </P>
        </div>
      </div>
      <div panel="panel2">
        <div>jajaja22</div>
        <div>
          <P>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec
            fermentum magna eu velit semper hendrerit. Aenean aliquet pretium
            justo, ut rhoncus magna euismod et. Donec condimentum congue sapien,
            nec auctor eros sagittis eget.
          </P>
          <P>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec
            fermentum magna eu velit semper hendrerit. Aenean aliquet pretium
            justo, ut rhoncus magna euismod et. Donec condimentum congue sapien,
            nec auctor eros sagittis eget.
          </P>
        </div>
      </div>
    </ShowCard>
  ));

/* 
  
  prototipo::

  - props:
    * singleDisplay: Solo una tab puede estar abierta al mismo tiempo
    * handleChange: Pasamos la funcion encargada de la apertura de tabs, para control condicional

  <ShowCard singleDisplay handleChange>
    <ShowCard.Title>
      titulo 1
    </ShowCard.Title>
    <ShowCard.Container>
      <P>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec
        fermentum magna eu velit semper hendrerit. Aenean aliquet pretium
        justo, ut rhoncus magna euismod et. Donec condimentum congue sapien,
        nec auctor eros sagittis eget.
      </P>
      <P>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec
        fermentum magna eu velit semper hendrerit. Aenean aliquet pretium
        justo, ut rhoncus magna euismod et. Donec condimentum congue sapien,
        nec auctor eros sagittis eget.
      </P>
    </ShowCard.Container>
  </ShowCard>
  
  */
