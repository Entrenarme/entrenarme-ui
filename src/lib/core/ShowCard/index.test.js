import * as React from 'react';
import { render } from 'react-testing-library';
import 'react-testing-library/cleanup-after-each';

import ShowCard from './index';

test('renders "Heading" with title', () => {
  const { container } = render(<ShowCard title="Mis cosas">Heading</ShowCard>);

  expect(container.firstChild).toMatchSnapshot();
});

test('renders "Heading" with no props', () => {
  const { container } = render(<ShowCard>Heading</ShowCard>);

  expect(container.firstChild).toMatchSnapshot();
});
