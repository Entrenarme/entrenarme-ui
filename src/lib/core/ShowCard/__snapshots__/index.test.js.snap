// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders "Heading" with no props 1`] = `
<div
  class="MuiPaper-root-61 MuiPaper-elevation1-64 MuiExpansionPanel-root-58 sc-bwzfXH loEQcT"
>
  <div
    tabindex="0"
    class="MuiButtonBase-root-94 MuiExpansionPanelSummary-root-88 sc-htpNat iUdPZr"
    role="button"
    aria-expanded="false"
  >
    <div
      class="MuiExpansionPanelSummary-content-92"
    >
      <h4
        class="sc-bdVaJa iRxabv"
        style="font-weight: 600; margin: 0px;"
      />
    </div>
    <div
      tabindex="-1"
      class="MuiButtonBase-root-94 MuiIconButton-root-97 MuiExpansionPanelSummary-expandIcon-93"
      role="button"
      aria-hidden="true"
    >
      <span
        class="MuiIconButton-label-102"
      >
        <svg
          aria-hidden="true"
          data-prefix="fas"
          data-icon="angle-down"
          class="svg-inline--fa fa-angle-down fa-w-10 "
          role="img"
          xmlns="http://www.w3.org/2000/svg"
          viewBox="0 0 320 512"
        >
          <path
            fill="currentColor"
            d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root-103"
      />
    </div>
  </div>
  <div
    class="MuiCollapse-container-110"
    style="min-height: 0px;"
    aria-hidden="true"
  >
    <div
      class="MuiCollapse-wrapper-112"
    >
      <div
        class="MuiCollapse-wrapperInner-113"
      >
        <div
          class="MuiExpansionPanelDetails-root-114 sc-bxivhb Ukcuw"
        >
          <div
            class="sc-ifAKCX eSrJSp"
          >
            Heading
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders "Heading" with title 1`] = `
<div
  class="MuiPaper-root-4 MuiPaper-elevation1-7 MuiExpansionPanel-root-1 sc-bwzfXH loEQcT"
>
  <div
    tabindex="0"
    class="MuiButtonBase-root-37 MuiExpansionPanelSummary-root-31 sc-htpNat iUdPZr"
    role="button"
    aria-expanded="false"
  >
    <div
      class="MuiExpansionPanelSummary-content-35"
    >
      <h4
        class="sc-bdVaJa iRxabv"
        style="font-weight: 600; margin: 0px;"
      >
        Mis cosas
      </h4>
    </div>
    <div
      tabindex="-1"
      class="MuiButtonBase-root-37 MuiIconButton-root-40 MuiExpansionPanelSummary-expandIcon-36"
      role="button"
      aria-hidden="true"
    >
      <span
        class="MuiIconButton-label-45"
      >
        <svg
          aria-hidden="true"
          data-prefix="fas"
          data-icon="angle-down"
          class="svg-inline--fa fa-angle-down fa-w-10 "
          role="img"
          xmlns="http://www.w3.org/2000/svg"
          viewBox="0 0 320 512"
        >
          <path
            fill="currentColor"
            d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root-46"
      />
    </div>
  </div>
  <div
    class="MuiCollapse-container-53"
    style="min-height: 0px;"
    aria-hidden="true"
  >
    <div
      class="MuiCollapse-wrapper-55"
    >
      <div
        class="MuiCollapse-wrapperInner-56"
      >
        <div
          class="MuiExpansionPanelDetails-root-57 sc-bxivhb Ukcuw"
        >
          <div
            class="sc-ifAKCX eSrJSp"
          >
            Heading
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
